api/test/trace/global_provider_test.cc

```diff
@@ -0,0 +1,34 @@
+#include "opentelemetry/trace/global_provider.h"
+
+#include <gtest/gtest.h>
+
+using opentelemetry::trace::Provider;
+using opentelemetry::trace::Tracer;
+
+class TestProvider : public opentelemetry::trace::TracerProvider
+{
+  Tracer *const GetTracer(opentelemetry::nostd::string_view library_name,
+                          opentelemetry::nostd::string_view library_version,
+                          const opentelemetry::trace::TracerOptions &tracer_options) override
+  {
+    return nullptr;
+  }
+};
+
+TEST(Provider, GetTracerProviderDefault)
+{
+  auto tf = Provider::GetTracerProvider();
+  ASSERT_NE(tf, nullptr);
+}
+
+TEST(Provider, SetTracerProvider)
+{
+  auto tf = new TestProvider();
+  // Capture the default provider constructed in order to free it
```

Comment:
I don't think we should be encouraging this.

IMO it'd be better to have SetTracerProvider (re)set back to the default provider so that this unit test doesn't permanently change global state (and thereby potentially introduce problems later where tests depend on execution order)